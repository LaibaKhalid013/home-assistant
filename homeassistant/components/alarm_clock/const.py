"""Constants for the Alarm Clock integration."""

import voluptuous as vol

from homeassistant.const import CONF_ICON, CONF_NAME
import homeassistant.helpers.config_validation as cv

DOMAIN = "alarm_clock"
ENTITY_ID_FORMAT = DOMAIN + ".{}"

ATTR_ALARM_TIME = "alarm_time"
ATTR_REPEAT_DAYS = "repeat_days"
ATTR_NEXT_ALARM = "next_alarm"

CONF_ALARM_TIME = "alarm_time"
CONF_REPEAT_DAYS = "repeat_days"

EVENT_ALARM_CLOCK_STARTED = "alarm_clock.started"
EVENT_ALARM_CLOCK_FINISHED = "alarm_clock.finished"
EVENT_ALARM_CLOCK_CANCELLED = "alarm_clock.cancelled"
EVENT_ALARM_CLOCK_CHANGED = "alarm_clock.changed"

STORAGE_KEY = DOMAIN
STORAGE_VERSION = 1

STORAGE_FIELDS = {
    vol.Required(CONF_NAME): cv.string,
    vol.Optional(CONF_ICON): cv.icon,
    vol.Required(CONF_ALARM_TIME): cv.time,
    vol.Required(CONF_REPEAT_DAYS): cv.weekdays,
}
