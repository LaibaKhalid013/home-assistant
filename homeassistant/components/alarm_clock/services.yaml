# Describes the format for available alarm clock services

turn_on:
  target:
    entity:
      domain: alarm_clock
  fields:
    alarm_time:
      example: "09:00:00"

turn_off:
  target:
    entity:
      domain: alarm_clock
