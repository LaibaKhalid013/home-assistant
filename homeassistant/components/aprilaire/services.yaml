set_dehumidity:
  target:
    entity:
      domain: climate
  fields:
    dehumidity:
      required: true
      selector:
        number:
          min: 10
          max: 50
          unit_of_measurement: "%"

trigger_air_cleaning_event:
  target:
    entity:
      domain: climate
  fields:
    event:
      required: true
      selector:
        select:
          options:
            - label: "3 Hours"
              value: "3hour"
            - label: "24 Hours"
              value: "24hour"

cancel_air_cleaning_event:
  target:
    entity:
      domain: climate

set_air_cleaning_mode:
  target:
    entity:
      domain: climate
  fields:
    mode:
      required: true
      selector:
        select:
          options:
            - label: "Off"
              value: "0"
            - label: Constant Clean
              value: "1"
            - label: Automatic
              value: "2"

toggle_air_cleaning_mode:
  target:
    entity:
      domain: climate
  fields:
    mode:
      required: true
      selector:
        select:
          options:
            - label: "Off"
              value: "0"
            - label: Constant Clean
              value: "1"
            - label: Automatic
              value: "2"

trigger_fresh_air_event:
  target:
    entity:
      domain: climate
  fields:
    event:
      required: true
      selector:
        select:
          options:
            - label: "3 Hours"
              value: "3hour"
            - label: "24 Hours"
              value: "24hour"

cancel_fresh_air_event:
  target:
    entity:
      domain: climate

set_fresh_air_mode:
  target:
    entity:
      domain: climate
  fields:
    mode:
      required: true
      selector:
        select:
          options:
            - label: "Off"
              value: "0"
            - label: Automatic
              value: "1"

toggle_fresh_air_mode:
  target:
    entity:
      domain: climate
  fields:
    mode:
      required: true
      selector:
        select:
          options:
            - label: "Off"
              value: "0"
            - label: Automatic
              value: "1"
