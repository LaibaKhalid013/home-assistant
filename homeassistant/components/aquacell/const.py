"""Constants for the Aquacell integration."""

DOMAIN = "aquacell"
DATA_AQUACELL = "DATA_AQUACELL"
