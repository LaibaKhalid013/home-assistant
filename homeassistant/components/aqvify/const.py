"""Constants for the Aqvify integration."""

import logging

DOMAIN = "aqvify"

CONF_DEVICE_KEY = "device_key"

LOGGER = logging.getLogger(__package__)
