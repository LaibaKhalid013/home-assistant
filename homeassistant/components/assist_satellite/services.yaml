play_media:
  target:
    entity:
      domain: assist_satellite
      supported_features:
        - assist_satellite.AssistSatelliteEntityFeature.ANNOUNCE
  fields:
    text:
      required: false
      example: "Time to wake up!"
      selector:
        text:
    media_id:
      required: false
      selector:
        text:
