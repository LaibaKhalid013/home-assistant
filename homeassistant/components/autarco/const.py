"""Constants for the Autarco integration."""

from __future__ import annotations

from datetime import timedelta
import logging
from typing import Final

DOMAIN: Final = "autarco"
LOGGER = logging.getLogger(__package__)
SCAN_INTERVAL = timedelta(minutes=1)

CONF_PUBLIC_KEY: Final = "public_key"

SENSORS_SOLAR: Final = "solar"
