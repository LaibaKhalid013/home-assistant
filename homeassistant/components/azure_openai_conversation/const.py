"""Constants for the OpenAI Conversation integration."""

import logging

DOMAIN = "azure_openai_conversation"
LOGGER = logging.getLogger(__package__)

CONF_AZURE_OPENAI_RESOURCE = "azure_openai_resource"
CONF_RECOMMENDED = "recommended"
CONF_PROMPT = "prompt"
CONF_CHAT_MODEL = "chat_model"
RECOMMENDED_CHAT_MODEL = "gpt-4"
CONF_MAX_TOKENS = "max_tokens"
RECOMMENDED_MAX_TOKENS = 150
CONF_TOP_P = "top_p"
RECOMMENDED_TOP_P = 1.0
CONF_TEMPERATURE = "temperature"
RECOMMENDED_TEMPERATURE = 1.0
AZURE_OPEN_API_VERSION = "2024-02-01"
