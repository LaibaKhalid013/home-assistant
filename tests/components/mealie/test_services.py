"""Tests for the Mealie services."""

from unittest.mock import AsyncMock

import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.mealie.const import ATTR_CONFIG_ENTRY_ID, DOMAIN
from homeassistant.components.mealie.services import SERVICE_GET_MEALPLAN_TODAY
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import ServiceValidationError

from tests.common import MockConfigEntry


async def test_service_mealplan_today(
    hass: HomeAssistant,
    mock_mealie_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test the get_mealplan_today service."""
    mock_config_entry.add_to_hass(hass)

    with pytest.raises(ServiceValidationError):
        await hass.services.async_call(
            DOMAIN,
            SERVICE_GET_MEALPLAN_TODAY,
            {ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id},
            blocking=True,
            return_response=True,
        )

    with pytest.raises(ServiceValidationError):
        await hass.services.async_call(
            DOMAIN,
            SERVICE_GET_MEALPLAN_TODAY,
            {ATTR_CONFIG_ENTRY_ID: "bad-config_id"},
            blocking=True,
            return_response=True,
        )

    assert await hass.config_entries.async_setup(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    response = await hass.services.async_call(
        DOMAIN,
        SERVICE_GET_MEALPLAN_TODAY,
        {ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id},
        blocking=True,
        return_response=True,
    )
    assert mock_mealie_client.get_mealplan_today.called
    assert response == snapshot
