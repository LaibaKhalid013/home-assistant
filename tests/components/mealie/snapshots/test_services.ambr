# serializer version: 1
# name: test_service_mealplan_today
  dict({
    <MealplanEntryType.BREAKFAST: 'breakfast'>: list([
    ]),
    <MealplanEntryType.DINNER: 'dinner'>: list([
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': datetime.date(2024, 1, 21),
        'mealplan_id': 192,
        'recipe': dict({
          'description': 'This is a wonderful option for picnics and grill outs when you are looking for a new take on potato salad. This simple side salad made with cauliflower, peas, and hard boiled eggs can be made the day ahead and chilled until party time!',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'qLdv',
          'name': 'Cauliflower Salad',
          'original_url': 'https://www.allrecipes.com/recipe/142152/cauliflower-salad/',
          'recipe_id': '40393996-417e-4487-a081-28608a668826',
          'recipe_yield': '6 servings',
          'slug': 'cauliflower-salad',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': datetime.date(2024, 1, 21),
        'mealplan_id': 206,
        'recipe': dict({
          'description': 'Easy, cheesy, sausage pasta! In the whirlwind of mid-week mayhem, dinner doesn’t have to be a chore – this 15-minute pasta, featuring HECK’s Chicken Italia Chipolatas is your ticket to a delicious and hassle-free mid-week meal.',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'BeNc',
          'name': '15 Minute Cheesy Sausage & Veg Pasta',
          'original_url': 'https://www.annabelkarmel.com/recipes/15-minute-cheesy-sausage-veg-pasta/',
          'recipe_id': '872bb477-8d90-4025-98b0-07a9d0d9ce3a',
          'recipe_yield': '',
          'slug': '15-minute-cheesy-sausage-veg-pasta',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': datetime.date(2024, 1, 21),
        'mealplan_id': 209,
        'recipe': dict({
          'description': '',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'ALqz',
          'name': 'Csirkés és tofus empanadas',
          'original_url': 'https://streetkitchen.hu/street-kitchen/csirkes-es-tofus-empanadas/',
          'recipe_id': '4233330e-6947-4042-90b7-44c405b70714',
          'recipe_yield': '16 servings',
          'slug': 'csirkes-es-tofus-empanadas',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': datetime.date(2024, 1, 21),
        'mealplan_id': 210,
        'recipe': dict({
          'description': 'This All-American beef stew recipe includes tender beef coated in a rich, intense sauce and vegetables that bring complementary texture and flavor.',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': '356X',
          'name': 'All-American Beef Stew Recipe',
          'original_url': 'https://www.seriouseats.com/all-american-beef-stew-recipe',
          'recipe_id': '48f39d27-4b8e-4c14-bf36-4e1e6497e75e',
          'recipe_yield': '6 servings',
          'slug': 'all-american-beef-stew-recipe',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.DINNER: 'dinner'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': datetime.date(2024, 1, 21),
        'mealplan_id': 223,
        'recipe': dict({
          'description': 'Jazz up chicken breasts in this fruity, sweetly spiced sauce with pomegranate seeds, toasted almonds and tagine paste',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'lF4p',
          'name': 'Pomegranate chicken with almond couscous',
          'original_url': 'https://www.bbcgoodfood.com/recipes/pomegranate-chicken-almond-couscous',
          'recipe_id': '27455eb2-31d3-4682-84ff-02a114bf293a',
          'recipe_yield': '4 servings',
          'slug': 'pomegranate-chicken-with-almond-couscous',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
    ]),
    <MealplanEntryType.LUNCH: 'lunch'>: list([
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.LUNCH: 'lunch'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': datetime.date(2024, 1, 21),
        'mealplan_id': 207,
        'recipe': dict({
          'description': '',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': None,
          'name': 'cake',
          'original_url': None,
          'recipe_id': '744a9831-fa56-4f61-9e12-fc5ebce58ed9',
          'recipe_yield': None,
          'slug': 'cake',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
      dict({
        'description': None,
        'entry_type': <MealplanEntryType.LUNCH: 'lunch'>,
        'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
        'mealplan_date': datetime.date(2024, 1, 21),
        'mealplan_id': 208,
        'recipe': dict({
          'description': 'Jazz up chicken breasts in this fruity, sweetly spiced sauce with pomegranate seeds, toasted almonds and tagine paste',
          'group_id': '0bf60b2e-ca89-42a9-94d4-8f67ca72b157',
          'image': 'lF4p',
          'name': 'Pomegranate chicken with almond couscous',
          'original_url': 'https://www.bbcgoodfood.com/recipes/pomegranate-chicken-almond-couscous',
          'recipe_id': '27455eb2-31d3-4682-84ff-02a114bf293a',
          'recipe_yield': '4 servings',
          'slug': 'pomegranate-chicken-with-almond-couscous',
          'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
        }),
        'title': None,
        'user_id': '1ce8b5fe-04e8-4b80-aab1-d92c94685c6d',
      }),
    ]),
    <MealplanEntryType.SIDE: 'side'>: list([
    ]),
  })
# ---
